content/spin/v2/redis-trigger.md

```diff
@@ -132,50 +126,29 @@ func main() {}
-
-The core Redis types are defined in [https://github.com/fermyon/spin/blob/main/wit/ephemeral/redis-types.wit](https://github.com/fermyon/spin/blob/main/wit/ephemeral/redis-types.wit), though note that not all of these are used in the pub-sub Redis trigger:
-
-<!-- @nocpy -->
-
-```fsharp
-// General purpose error.
-enum error {
-    success,
-    error,
-}
-
-// The message payload.
-type payload = list<u8>
-
-// Remaining types are not used in the trigger
-```
-
-> The same Redis types are also used to model the API for sending outbound Redis requests.
+The Redis component interface is defined using a WebAssembly Interface (WIT) file.  ([Learn more about the WIT language here.](https://component-model.bytecodealliance.org/design/wit.html)).  You can find the latest WITs for Spin Redis components at [https://github.com/fermyon/spin/tree/main/wit/preview2](https://github.com/fermyon/spin/tree/main/wit/preview2).
 
-The entry point for Spin Redis components is then defined in [https://github.com/fermyon/spin/blob/main/wit/ephemeral/spin-redis.wit](https://github.com/fermyon/spin/blob/main/wit/ephemeral/spin-redis.wit):
+In particular, the entry point for Spin Redis components is defined in [the `inbound-redis` interface](https://github.com/fermyon/spin/blob/main/wit/preview2/deps/spin%40unversioned/inbound-redis.wit):
 
 <!-- @nocpy -->
 
 ```fsharp
-// wit/ephemeral/spin-redis.wit
-
-use * from redis-types
+interface inbound-redis {
+  use redis-types.{payload, error}
 
-// The entry point for a Redis handler.
-handle-redis-message: func(message: payload) -> expected<unit, error>
+  // The entrypoint for a Redis handler.
+  handle-message: func(message: payload) -> result<_, error>
+}
 ```
 
-This is the function signature that all Redis components must implement, and
+This is the interface that all Redis components must implement, and
 which is used by Spin when instantiating and invoking the component.
 
-This interface (`spin-redis.wit`) can be directly used together with the
-[Bytecode Alliance `wit-bindgen` project](https://github.com/bytecodealliance/wit-bindgen)
-to build a component that Spin can invoke.
+However, this is not necessarily the interface you, the component author, work with. It may not even be the interface of the component you build!
+
+In many cases, you will use a more idiomatic wrapper provided by the Spin SDK, which implements the "true" interface internally. In some cases, you will build a Wasm "core module" which implements an earlier version of the Spin Redis interface, which Spin internally adapts to the "true" interface as it loads your module.
 
-This is exactly how Spin SDKs, such as the [Rust](rust-components) and [Go](go-components) SDKs, are built.
-As more languages add support for the component model, we plan to add support for them in the same way.
+But if you wish, and if your language supports it, you can implement the `inbound-redis` interface directly, using tools such as the
```

Comment:
Okay, this might not be true - I found a PR comment saying you _won't_ be able to use CM tooling for the Redis trigger - awaiting confirmation.

Reply:
We are not encouraging people to do this so I have changed it to mention the interface but not the "hey let's write the SDK right here in the barn" stuff.